feat!: refactor module

mfroembgen opened the pull request.

renovate[bot] committed:
chore(deps): update terraform justtrackio/dynamodb-table/aws to v1.0.4

mfroembgen committed:
feat!: refactor_module

mfroembgen committed:
feat!: refactor_module

applike-ss commented on main.tf:
```diff
@@ -1,227 +1,45 @@
-  count   = var.redis_enabled ? 1 : 0
+module "kvstore_label" {
   source  = "justtrackio/label/null"
   version = "0.26.0"
```
please make sure we can enable/disable the redis/ddb independently

mfroembgen committed:
feat!: refactor_module

mfroembgen committed:
feat!: refactor_module

mfroembgen committed:
feat!: refactor_module

applike-ss commented on examples/all-labels/main.tf:
```diff
@@ -1,19 +0,0 @@
-module "example" {
```
why did you remove all examples?

mfroembgen replied:
Because it will still change a bit and I don't want to waste now time until I know it will stay like that for quite some time

applike-ss commented on variables.tf:
```diff
@@ -1,21 +1,7 @@
-variable "ddb_autoscale_read_schedule" {
```
why did you remove the ddb scaling schedules?

mfroembgen replied:
currently its not supported

mfroembgen replied:
I do need to add a pr for that

applike-ss commented on main.tf:
```diff
@@ -1,227 +1,54 @@
+  source = "github.com/justtrackio/terraform-aws-ecs-redis?ref=add_service_discovery_name_varibale"
+
+  context      = module.kvstore_label.context
+  label_orders = var.label_orders
```
assuming we have a label order of `["environment","stage","name"]`
wouldn't that result in this dynamodb table name `env-stg-name`, while the redis' ecs service is going to be also called like that?
In my opinion, this doesn't make sense, when we have a cluster name of `env`.
Shouldn't we revert to the label order per service again here?

mfroembgen replied:
It uses the label_orders ecs not kvstore. I put it inside this module as well so that we can override it if we want.

applike-ss commented on main.tf:
```diff
@@ -1,227 +1,54 @@
-  port_mappings = [
+module "ddb" {
+  count   = var.ddb_enabled ? 1 : 0
+  source  = "terraform-aws-modules/dynamodb-table/aws"
```
why are you switching away from our own module to this one?

mfroembgen replied:
In this case for consistency as I don't want to use multiple different ddb modules.

mfroembgen committed:
feat!: refactor_module

mfroembgen committed:
feat!: refactor_module

ztzxt committed:
feat: add ddb autoscaling max capacity (#37)

## Description
<!--- Describe your changes in detail -->

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

## Breaking Changes
<!-- Does this break backwards compatibility with the current major
version? -->
<!-- If so, please provide an explanation why it is necessary. -->

## How Has This Been Tested?
- [ ] I have updated at least one of the `examples/*` to demonstrate and
validate my change(s)
- [ ] I have tested and validated these changes using one or more of the
provided `examples/*` projects
<!--- Users should start with an existing example as its written, deploy
it, then check their changes against it -->
<!--- This will highlight breaking/disruptive changes. Once you have
checked, deploy your changes to verify -->
<!--- Please describe how you tested your changes -->
- [ ] I have executed `pre-commit run -a` on my pull request
<!--- Please see
https://github.com/antonbabenko/pre-commit-terraform#how-to-install for
how to install -->

ztzxt committed:
fix: run pre-commit hooks
